src/facilities/costCenters.spec.js

```diff
@@ -0,0 +1,548 @@
+      let request;
+
+      beforeEach(function() {
+        expectedCostCenterFacility = fixture.build('costCenterFacility');
+        expectedFacilityId = expectedCostCenterFacility.facilityId;
+        expectedCostCenterId = expectedCostCenterFacility.costCenterId;
+        rawCostCenterFacility = fixture.build('costCenterFacility', null, {
+          fromServer: true
+        });
+
+        formatCostCenterFacilityFromServer = this.sandbox
+          .stub(facilitiesUtils, 'formatCostCenterFacilityFromServer')
+          .returns(expectedCostCenterFacility);
+        request = {
+          ...baseRequest,
+          post: this.sandbox.stub().resolves(rawCostCenterFacility)
+        };
+
+        const costCenters = new CostCenters(baseSdk, request, expectedHost);
+
+        promise = costCenters.addFacility(
+          expectedCostCenterId,
+          expectedFacilityId
+        );
+      });
+
+      it('creates the new cost center <--> facility relationship', function() {
+        expect(request.post).to.be.calledWith(
+          `${expectedHost}/costcenters/${expectedCostCenterId}/facility/${expectedFacilityId}`
+        );
+      });
+
+      it('formats the returning cost center facility object', function() {
+        return promise.then(() => {
+          expect(formatCostCenterFacilityFromServer).to.be.calledWith(
+            rawCostCenterFacility
+          );
+        });
+      });
+
+      it('returns a fulfilled promise with the new facility information', function() {
+        return expect(promise).to.be.fulfilled.and.to.eventually.equal(
+          expectedCostCenterFacility
+        );
+      });
+    });
+
+    context('when there is missing required information', function() {
+      ['costCenterId', 'facilityId'].forEach(function(field) {
+        it(`it throws an error when ${field} is missing`, function() {
+          const costCenterFacility = fixture.build('costCenterFacility');
+          delete costCenterFacility[field];
+
+          const costCenters = new CostCenters(
+            baseSdk,
+            baseRequest,
+            expectedHost
+          );
+          const promise = costCenters.addFacility(
+            costCenterFacility.costCenterId,
+            costCenterFacility.facilityId
+          );
+
+          return expect(promise).to.be.rejectedWith(
+            `A ${field} is required to create a relationship between a cost center and a facility.`
+          );
+        });
+      });
+    });
+  });
+
+  describe('create', function() {
+    context('when all required information is supplied', function() {
+      let formatCostCenterFromServer;
+      let formatCostCenterToServer;
+      let formattedCostCenterFromServer;
+      let formattedCostCenterToServer;
+      let initialCostCenter;
+      let promise;
+      let request;
+
+      beforeEach(function() {
+        initialCostCenter = fixture.build('costCenter');
+        formattedCostCenterToServer = fixture.build('costCenter', null);
```

Comment:
Minor quibble, but can you get rid of `null` here? (Will be clearer in the future instead of someone looking at this and wondering why it is needed)